tracing/trace_state.cc

```diff
@@ -163,6 +163,14 @@ void trace_state::add_prepared_query_options(const cql3::query_options& prepared
     }
 }
 
+static sstring param_key(const char *key, size_t index, size_t size) {
```

Comment:
This helper function will still allocate, and its return value would be used later in a `format()` call inside `build_parameters_map_for_one_prepared`, which means we allocate a string twice for a single purpose. It would be more optimal to modify `build_parameters_map_for_one_prepared` itself to take the index and call the formatting once, since this series is about reducing allocations.

Reply:
While you are not wrong, @psarna, Ivan is right in that at this point in the flow we prefer code readability to performance because this is NOT a "hot path".
Parameters are built when we are going to write the tracing data to the I/O backend and these events are supposed to be rare.